### DISCUSSION TITLE
I get some special masks by expanding the character masks. Why do I get gray areas when I use these masks for inpaint

### DISCUSSION THREAD

**Discussion by RonnySS** [lllyasviel/Fooocus#2807] [Q&A]
This mask is obtained by first identifying the characters with EfficientSAM, then doing some mask extensions, What I use is Fooocus - API project (https://github.com/mrhan1993/Fooocus-API) or Fooocus inpaint Node (https://github.com/Acly/comfyui-inpaint-nodes), so I can be introduced into a mask when inpaint instead of manual blotted out. But the results are similar to the figure below, and this phenomenon occurs in many other roles
![expand_mask](https://github.com/lllyasviel/Fooocus/assets/47497909/437e932b-665f-4414-a170-31aedaf9287a)

and I used a very general set of parameters
![Prompt](https://github.com/lllyasviel/Fooocus/assets/47497909/c5aff023-6160-4ea0-a42c-e475567664c7)

This is the mask,  it has a lot of jagged. At first I thought it was a sawtooth problem, until I used some anti-aliasing or feathering to make masks that still had gray edges
![全部外扩去除部分](https://github.com/lllyasviel/Fooocus/assets/47497909/beb2e0f9-41d7-45e6-8cc0-7f6dffc97404)

I also made a few other attempts, such as taking just a small area of inpaint, or panning it to see the effect
![regional_mask_translation_right](https://github.com/lllyasviel/Fooocus/assets/47497909/72b93520-6c41-48e1-9860-e2d1ff5998ea)

![regional_mask_translation_left](https://github.com/lllyasviel/Fooocus/assets/47497909/d1557086-1761-4b69-912d-217f8324d623)

and the most strange thing is, when I try to simulate this mask by hand in fooocus, Everything's fine again
![image](https://github.com/lllyasviel/Fooocus/assets/47497909/3ea0d8ab-e0c4-4d9b-a006-f6f8467eb394)

This problem has puzzled me for a lot of time, I have tried hundreds of diagrams, and also tried various bottom molds, including sd_xl_base1.0. At present, I find that inpaint is far enough away from characters to avoid it. Someone can tell me the possible reasons for this phenomenon or the principle of inpaint algorithm of Fooocus is similar to other algorithms. I would like to solve it or avoid it, because in addition to this problem, inpaint algorithm of Fooocus is really really really good for me

**Comment by mashb1t** [comment]
@RonnySS thank you for the well-written message, much appreciated. Before potentially going into more detail, can you please provide more information about the inpaint method and prompt you chose to create above images? (default, detail, modify + prompt if any)

**Reply by BNP1111** [reply]
I have always believed that the "Fooocus" node in ComfyUI, along with other third-party implementations of Fooocus inpaint, cannot match the original Fooocus. There are undoubtedly significant differences in code logic and some hidden techniques, which is why I have consistently stuck with using the original Fooocus for inpainting instead of ComfyUI.

**Reply by RonnySS** [reply]
Thank you for your reply, my inpaint method is **default**, prompt is **night, trees, leaves, grass, lawn \ n trees**. I'll be taking a closer look at the Fooocus - API project to see if there are any problem with it
